packages/qwik-city/runtime/src/qwik-image-component.tsx

```diff
@@ -0,0 +1,291 @@
+import {
+  component$,
+  createContextId,
+  useContext,
+  QRL,
+  QwikIntrinsicElements,
+  useBrowserVisibleTask$,
+  useContextProvider,
+  useSignal,
+  useTask$,
+} from '@builder.io/qwik';
+
+export const DEFAULT_RESOLUTIONS = [
+  6016, // 6K
+  5120, // 5K
+  4480, // 4.5K
+  3840, // 4K
+  3200, // QHD+
+  2560, // WQXGA
+  2048, // QXGA
+  1920, // 1080p
+  1668, // Various iPads
+  1280, // 720p
+  1080, // iPhone 6-8 Plus
+  960, // older horizontal phones
+  828, // iPhone XR/11
+  750, // iPhone 6-8
+  640, // older and lower-end phones
+];
+
+export const getBreakpoints = ({
+  width: widthAttribute,
+  layout,
+  resolutions = [],
+}: Pick<QwikImageProps, 'width' | 'layout' | 'resolutions'>): number[] => {
+  if (layout === 'fullWidth') {
+    return resolutions;
+  }
+  if (!widthAttribute) {
+    return [];
+  }
+  const width = typeof widthAttribute === 'string' ? parseInt(widthAttribute, 10) : widthAttribute;
+  const doubleWidth = width * 2;
+  if (layout === 'fixed') {
+    return [width, doubleWidth];
+  }
+  if (layout === 'constrained') {
+    return [
+      // Always include the image at 1x and 2x the specified width
+      width,
+      doubleWidth,
+      // Filter out any resolutions that are larger than the double-res image
+      ...resolutions.filter((w) => w < doubleWidth),
+    ];
+  }
+
+  return [];
+};
+
+export const getSrcSet = ({
+  src,
+  width,
+  height,
+  aspectRatio,
+  resolutions,
+  layout,
+}: Pick<
+  QwikImageProps,
+  'src' | 'width' | 'height' | 'aspectRatio' | 'resolutions' | 'layout'
+>): string => {
+  const breakpoints = getBreakpoints({
+    width: typeof width === 'string' ? parseInt(width, 10) : width,
+    layout,
+    resolutions: resolutions || DEFAULT_RESOLUTIONS,
+  });
+  return breakpoints
+    .sort()
+    .map((bp) => {
+      let transformedHeight;
+      if (height && aspectRatio) {
+        transformedHeight = Math.round(bp * aspectRatio);
+      }
+
+      const transformed = {
+        url: src,
+        width: bp,
+        height: transformedHeight,
+      };
+
+      if (transformed) {
+        return `${transformed.url}?w=${bp} ${bp}w`;
```

Comment:
Can this not use the same loader as the blur image?

Reply:
I pushed a version without blur feature. let's keep it simple for the moment